### PR TITLE
use GetInstanceProviderID to get instance provider ID

### PR DESCRIPTION
**Pull request by gongguan**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
fix https://github.com/kubernetes/kubernetes/pull/94846#discussion_r490974724

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go [hunk 1]
```diff
@@ -264,7 +264,7 @@ func (g *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error)
 	providerID := node.Spec.ProviderID
 	if providerID == "" {
 		var err error
-		if providerID, err = g.InstanceID(ctx, types.NodeName(node.Name)); err != nil {
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go:267]: I think InstanceID itself is expected to return the InstanceNotFound error so there's a potential optimization here to return `false, nil` early if InstanceID returned InstanceNotFound.

**Review by andrewsykim — COMMENTED** [review]
/assign @cheftako @wojtek-t

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go (hunk 1 above)
Comment (gongguan) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go:267]: done

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go [hunk 2]
```diff
@@ -264,7 +264,10 @@ func (g *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error)
...
+		if providerID, err = cloudprovider.GetInstanceProviderID(ctx, g, types.NodeName(node.Name)); err != nil {
```
Comment (wojtek-t) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go:267]: This looks fine to me, but given we merged with this bug, this deserves some unit test.

Can you please add one?

Comment (gongguan) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go:267]: Done, @wojtek-t PTAL.

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		},
+		{
+			name:        "node not exist",
+			nodeName:    "test-node-2",
+			exist:       false,
+			expectedErr: fmt.Errorf("failed to get instance ID from cloud provider: instance not found"),
+		},
+	}
+
+	for _, test := range testcases {
```
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances_test.go:61]: wrap this with a t.Run
```
for _, test := range testcases {
  t.Run(test.name, func(t *testing.T) {
     ...
  }) 
}
```

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go (hunk 2 above)
Comment (gongguan) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go:267]: Done, PTAL.

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances_test.go (hunk 3 above)
Comment (gongguan) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances_test.go:61]: Done.

**Comment by wojtek-t** [comment]
@gongguan - please squash the commits (other than that lgtm)

**Comment by gongguan** [comment]
@wojtek-t Does second commit which fix `createAndInsertNodes` worth a separated commit?

**Comment by wojtek-t** [comment]
> @wojtek-t Does second commit which fix createAndInsertNodes worth a separated commit?

I would squash it too.

**Comment by gongguan** [comment]
@wojtek-t squashed, PTAL.

**Pull request merged**